alfresco-search/src/main/java/org/alfresco/solr/AlfrescoCoreAdminHandler.java

```diff
@@ -838,6 +843,153 @@ private void actionREPORT(SolrQueryResponse rsp, SolrParams params, String cname
         }
     }
 
+    private void rangeCheck(SolrQueryResponse rsp,String cname) throws IOException
+    {
+        InformationServer informationServer = informationServers.get(cname);
+
+        Collection<Tracker> trackers = trackerRegistry.getTrackersForCore(cname);
+        MetadataTracker metadataTracker = null;
+        for(Tracker tracker : trackers)
+        {
+            if(tracker instanceof MetadataTracker)
+            {
+                metadataTracker = (MetadataTracker)tracker;
+            }
+        }
+
+        DocRouter docRouter = metadataTracker.getDocRouter();
+
+        if(docRouter instanceof DBIDRangeRouter) {
+
+            DBIDRangeRouter dbidRangeRouter = (DBIDRangeRouter) docRouter;
+            long startRange = dbidRangeRouter.getStartRange();
+            long endRange = dbidRangeRouter.getEndRange();
+
+            long maxNodeId = informationServer.maxNodeId();
+            long nodeCount = informationServer.nodeCount();
+
+            long bestGuess = -1;  // -1 means expansion cannot be done. Either because expansion
+                                 // has already happened or we're above safe range
+
+            long range = endRange - startRange; // We want this many nodes on the server
+            //System.out.println("##### Range:" + range);
+
+            long midpoint = startRange + ((long) (range * .5));
+            //System.out.println("##### Midpoint:"+midpoint);
+
+            long safe = startRange + ((long) (range * .75));
+            //System.out.println("##### Safe:"+safe);
+
+
+            long offset = maxNodeId-startRange;
+            //System.out.println("#####offset:"+offset);
+
+            //System.out.println("#####max nodeid:"+maxNodeId);
+
+            double density = 0;
+
+            if(offset > 0) {
+                density = ((double)nodeCount) / ((double)offset); // This is how dense we are so far.
+            }
+
+            //System.out.println("#####density:"+density);
+
+            if (!dbidRangeRouter.getExpanded())
+            {
+                if(maxNodeId <= safe)
+                {
+                    if (maxNodeId >= midpoint)
+                    {
+                        if(density >= 1)
+                        {
+                            //This is fully dense shard. I'm not sure if it's possible to have more nodes on the shards
+                            //then the offset, but if it does happen don't expand.
+                            bestGuess=0;
+                        }
+                        else
+                        {
+                            double multiplier = range / (density*range); // Extrapolate across the entire range to find the multiplier

```

Comment:
multiplier = 1.0 / density;

Reply:
Yes, this is simpler. Is what I have equivalent? I'll change and see what happens with the tests. If the test break then they are not equivalent.

Reply:
Tests still pass after making this change so they were equivalent. I'll go with the suggested formula though as its simpler.